### ISSUE TITLE
HDR MergeMertens - tif uint16 output missing

### ISSUE THREAD

**Issue by mattiik** [opencv/opencv#13296]
When using python to produce LDR image (example [here](https://docs.opencv.org/3.2.0/d3/db7/tutorial_hdr_imaging.html)) I am able to import TIFF 16 bit , three channel , colour images unchanged. However processing using MergeMertens results only in uint 8-Bit tiff colour images.

Would it be possible to include something like bittype as an argument when using mergemertens?
I could not find any option (or documentation) how to solve this problem.

- OpenCV => 3.2.0
- Operating System / Platform => Windows 64 Bit
- Python=> 2.7.15

**Comment by alalek** [comment]
`MergeMertens` output is `CV_32F`.
Perhaps your problem is related to TIFF encoder (it doesn't expect CV_32F data and converts it into `CV_8U`). You can try to convert it manually into `CV_16S`/`CV_16U` manually before calling `imwrite()`.

**Comment by mattiik** [comment]
Hi @alalek ,

Thank you for the prompt answer. I'm sorry that there was a misunderstanding of the question:
I wish to use several 16 bit TIF images as INPUT to MergeMertens.
In the MergeMertens routine it is hard coded that the input files are 8 bit, since the number 255 occurs explicitely in the corresponding code.
Might it be possible to change this into a variable that can also assume the value 65535 for 16 bit input?

**Comment by alalek** [comment]
Do you mean this [line](https://github.com/opencv/opencv/blob/3.4.4/modules/photo/src/merge.cpp#L180) ?

**Comment by mattiik** [comment]
Exactly! It divides each pixel  by the maximum to get a float in [0,1].
However, if the true maximum is 65535 it doesn't work. With a variable maximum as argument the algorithm should work fine, shouldn't it?

**Comment by alalek** [comment]
Great!

Looks like current implementation assumes `CV_8U` input only (but it continues to work with other types too - due missing check).
I believe we should:
- assume `CV_32F` input type normalized (in the range 0..1 and don't apply scale)
- forbid other types than `CV_8U` / `CV_32F`.

`CV_16U` input should be normalized before calling of implementation:
Workaround: `src[i].convertTo(src[i], CV_32F, 1.0f/65535.0f)`

Workaround for OpenCV <= 3.4.4 or OpenCV 4.0 with `CV_16U` input:
```
src[i].convertTo(src[i], CV_32F, 255.0f/65535.0f)
```

BTW, There is [base class `MergeExposures`](https://docs.opencv.org/3.4/d6/d5e/classcv_1_1MergeExposures.html#a547fe61891e611804b27a3f5ef0237ee) and it assumes CV_8U images only (see the "response" parameter - it has 256x1 layout).
